lib/rspec/support/differ.rb

```diff
@@ -116,6 +115,12 @@ def build_hunks(actual, expected)
         HunkGenerator.new(actual, expected).hunks
       end
 
+      def final_line(last_hunk)
+        output_message = last_hunk.diff(format_type).to_s.strip
+        # Checks last char of string for + or -, appends message on match
+        output_message.gsub(/(?<=[+-]$)\z/, "\\ No newline at end of input")
```

Comment:
Something along the lines of:

```Ruby
def expected_lines
  split_into_lines(@expected)
end

def actual_lines
  split_into_lines(@actual)
end

private

def split_into_lines(string)
  string << ''\ No newline at end of input" if string[-1] != "\n"
  string.split("\n").map! { |e| e.chomp }
end
```

This is psuedo code though, I haven't tried it directly